README.md

```diff
@@ -1,36 +1,68 @@
+Use the `ExecuteOnReload` attribute on static methods that you want to execute when entering play mode with domain reloading disabled.
 
 ### Examples
 ```csharp
 public class CharacterManager : MonoBehaviour
 {
-  //Will set value to default (null)
+  // Will set value to default (null).
   [ClearOnReload]
   static CharacterManager instance;
   
-  //Works on events!
-  [ClearOnReload]
-  static event Action onDoSomething;
-  
-  //Will set value to defined value (10)
+  // Will set variable to given value (10).
   [ClearOnReload(valueToAssign=10)]
   static int startsAsTen;
   
-  //Will reset value, but make it not null (creates a new instance)
+  // Will reset value, creating a new instance using default constructor.
   [ClearOnReload(assignNewTypeInstance=true)]
   static CharacterManager myNeverNullManager;
   
-  [ExecuteOnReloadAttribute]
+  // Will execute this method.
+  [ExecuteOnReload]
   static void RunThis() 
   {
-      Debug.Log("Clean up here or something")
+    Debug.Log("Clean up here.")
+  }
+
+  // Does not work on events!
+  // [ClearOnReload] 
+  static event Action onDoSomething;
+  
+  // However, one can use ExecuteOnReload to do their own clean up.
+  [ExecuteOnReload]
+  static void CleanUpEvents() 
+  {
+    foreach(Delegate d in onDoSomething.GetInvocationList())
+      onDoSomething -= d;
+   
   }
 }
 ```
+
+## Changes
+
+On my game project which admittedly has a lot of assemblies, I added a profile statement to the [original project](https://github.com/joshcamas/unity-domain-reload-helper) and [it took 5 seconds to run](https://github.com/shanecelis/UnityDomainReloadHelper/blob/master/Documentation%7E/original-performance.png) the `DomainReloadHandler` code, which virtually nullifies the gains from turning off domain reloading. So it made the allure of these convenience attributes impractical.
+
+Alexey Zakharov [mentioned using TypeCache in Steinhauer's
+post](https://forum.unity.com/threads/attribute-to-clear-static-fields-on-play-start.790226/#post-5262665).
+In a quick hack, I used [TypeCache](https://docs.unity3d.com/ScriptReference/TypeCache.html) and instead of `DomainReloadHandler` taking 5 seconds, [it took 46
+ms](https://github.com/shanecelis/UnityDomainReloadHelper/blob/master/Documentation%7E/modified-performance.png). 
+
+There are some disadvantages however. TypeCache only supports finding fields, methods, types, and derived types. But I think speed is a worthwhile advantage.
+
+## License
+
+This project is released under the MIT license.
+
+## Acknowledgments
+
+The [original project](https://github.com/joshcamas/unity-domain-reload-helper) was written by Josh Steinhauer. Kudos on the design. I didn't know about TypeCache either, and I wrote similar assembly walking code that's still present in [Minibuffer Console](http://seawisphunter.com/products/minibuffer/).
+
+[^1]: This is not the original project. It was forked by me. See Changes section for more details.
```

Comment:
Wanna remove this last line for the merge?

Reply:
Happily. I adjusted the README to fit with the original.